smack-extensions/src/main/java/org/jivesoftware/smackx/usertune/element/UserTuneElement.java

```diff
@@ -0,0 +1,271 @@
+        return message.hasExtension(ELEMENT, NAMESPACE);
+    }
+
+    public static UserTuneElement from(Message message) {
+        return message.getExtension(ELEMENT, NAMESPACE);
+    }
+
+    @Override
+    public int hashCode() {
+        return HashCode.builder()
+                .append(artist)
+                .append(length)
+                .append(rating)
+                .append(source)
+                .append(title)
+                .append(track)
+                .append(uri).build();
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        return EqualsUtil
+                .equals(this, obj, (equalsBuilder, otherTune) -> equalsBuilder
+                                                            .append(artist, otherTune.artist)
+                                                            .append(length, otherTune.length)
+                                                            .append(rating, otherTune.rating)
+                                                            .append(source, otherTune.source)
+                                                            .append(title, otherTune.source)
+                                                            .append(title, otherTune.title)
+                                                            .append(track, otherTune.track)
+                                                            .append(uri, otherTune.uri));
+    }
+
+    /**
+     * Returns a new instance of {@link Builder}.
+     * @return Builder
+     */
+    public static Builder getBuilder() {
+        return new Builder();
+    }
+
+    /**
+     * This class defines a Builder class for {@link UserTuneElement}. <br>
+     * {@link UserTuneElement} instance can be obtained using the {@link #build()} method as follows. <br>
+     * UserTuneElement.Builder builder = new UserTuneElement.Builder();
+     * builder.setSource("Yessongs"); builder.setTitle("Heart of the Sunrise");
+     * UserTuneElement userTuneElement = builder.build(); <br>
+     * Values such as title, source, artist, length, source, track and uri can be set using their respective setters through {@link Builder}
+     */
+    public static final class Builder {
+        private String artist;
+        private UInt16 length;
+        private Integer rating;
+        private String source;
+        private String title;
+        private String track;
+        private URI uri;
+
+        private Builder() {
+        }
+
+        /**
+         * Artist is an optional element in UserTuneElement.
+         * @param artist
+         * @return Builder
+         */
+        public Builder setArtist(String artist) {
+            this.artist = artist;
+            return this;
+        }
+
+        /**
+         * Length is an optional element in UserTuneElement.
+         * @param length
+         * @return Builder
+         */
+        public Builder setLength(int length) {
+            this.length = UInt16.from(length);
```

Comment:
Should be `return setLength(UInt16.from(length))`, so that the actually setting of the length is delegated to a single method.